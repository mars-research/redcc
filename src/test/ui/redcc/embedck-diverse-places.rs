// build-pass

#![feature(rustc_attrs)]

#[cfg_attr(not(test), rustc_diagnostic_item = "RRef")]
struct RRef<T>(T);
struct Direct(RRef<i32>);

fn main() {
    // place is array
    let mut array = RRef([RRef(0), RRef(1)]);
    array.0[0] = RRef(2);

    let mut array = [RRef(0), RRef(1)];
    array[0] = RRef(2);

    // tuple
    let mut tuple = RRef((RRef(0), RRef(1)));
    tuple.0.0 = RRef(2);

    let mut tuple = (RRef(0), RRef(1));
    tuple.0 = RRef(2);

    // reference
    let mut r = &mut RRef(RRef(1));
    r.0 = RRef(0);

    let mut r = &mut Direct(RRef(1));
    r.0 = RRef(0);
}
