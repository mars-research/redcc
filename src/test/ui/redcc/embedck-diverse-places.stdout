EMBED RRef<i32> -> RRef<[RRef<i32>; 2]>
EMBED RRef<i32> -> RRef<(RRef<i32>, RRef<i32>)>
EMBED RRef<i32> -> &mut RRef<RRef<i32>>
